nucypher/blockchain/eth/actors.py

```diff
@@ -266,6 +271,7 @@ def _make_condition_provider(uri: str) -> HTTPProvider:
         provider = HTTPProvider(endpoint_uri=uri)
         return provider
 
+
     def connect_condition_providers(
         self, endpoints: Dict[int, List[str]]
     ) -> DefaultDict[int, Set[HTTPProvider]]:
```

Comment:
The creation of this set of providers is only done once on startup. Would checking health need to be ongoing/periodic? When evaluating conditions, I know the node iterates through the set of endpoints if there are multiple available and if prior ones experience execution issues/failures, but should we adjust the order based on problematic endpoints?

For example, say on startup we have 10 endpoints that are healthy, so we always use the 1st endpoint - no problem. However, 2 days later, the first 5 endpoints are now unhealthy; then, for subsequent condition evaluation, we will always iterate through and try the first 5, with each experiencing failure before we move to the next and eventually use the valid 6th endpoint. This will continue until either:
1. The earlier endpoints in the first 5 spots resolve their issues
2. The node restarts

In this way, there may be an efficiency hit if the first 5 always have to fail before we use the 6th. This is probably more pronounced if default RPC endpoints are earlier in the set (e.g., overloading of public endpoints)—see my prior comment about whether we want to bias towards user-defined endpoints.

Reply:
This is something that has crossed my mind but I have been uncertain about when to perform a refresh.  In fact this can be further broken down into smaller pieces. (1) Refreshing the chainlist/endpoints URLs and (2) performing an additional round of health checks.  It will be good to take an incremental approach, but ideally it will be good if the user-defined endpoints can be always attempted first.